src/protocols/vnc/user.c

```diff
@@ -112,6 +112,9 @@ int guac_vnc_user_leave_handler(guac_user* user) {
 
     guac_vnc_client* vnc_client = (guac_vnc_client*) user->client->data;
 
+    if (!vnc_client || !user || !vnc_client->display || !vnc_client->display->cursor)
```

Comment:
Summarizing [my comment from earlier](https://github.com/apache/incubator-guacamole-server/pull/109#issuecomment-331722195), this check seems overly broad, and it's unlikely that all of these pointers need to be verified. We shouldn't be paranoidly checking for `NULL` every time we're about to use a pointer.

If we're going to add checks, the reason for each check should be known, and the cause of [GUACAMOLE-388](https://issues.apache.org/jira/browse/GUACAMOLE-388) needs to be determined.

Reply:
hello, I'm doing this in my spare time.

I do not mind if you want to determine the root cause of GUACAMOLE-388

what I wanted to do is to add check for null, it still makes sense, i.e. "check for null before dereferencing the pointer". it will help if something will change later on the caller side.


as for "overly broad", I'll make it more narrow soon

Reply:
> I do not mind if you want to determine the root cause of GUACAMOLE-388

Excellent, as that's really the first step before attempting any sort of fix. It's not a good idea to simply cast a wide net without knowing the actual cause, and then call it fixed when the problem cannot be reproduced.

> what I wanted to do is to add check for null, it still makes sense, i.e. "check for null before dereferencing the pointer". it will help if something will change later on the caller side.

No, blanket/paranoid checks are not good practice. We shouldn't be checking for `NULL` simply as a matter of course. Things should be written purposefully, and we should know what we're passing in. Coding defensively like that is self-defeating.